### PR TITLE
Add examples for the PSA Crypto API

### PR DESCRIPTION
**Pull request by hannestschofenig**
## Description

This PR contributes examples to help developers better understand the PSA Crypto API.

## Status

**IN DEVELOPMENT**

## Requires Backporting

NO  

## Migrations

NO

## Additional comments
Any additional information that could be of interest

## Todos
- [ ] Documentation
- [ ] Changelog updated

## Steps to test or reproduce

Just run the educational examples

### CODE REVIEW

**Comment by mprse** [comment]
What is current status of this PR?

**Comment by hannestschofenig** [comment]
The status of this PR is the following: While I am still fine-tuning the examples I believe it should otherwise be ready for inclusion.

**Comment by gilles-peskine-arm** [comment]
@hannestschofenig Thanks for these examples. But before we review, please fix the issues reported by the CI. Let us know if you need help understanding some of the errors.

**Comment by tom-daubney-arm** [comment]
All internal CI is passing. OpenCI errors are due to a known issue with the CI that is being worked on.

File: library/ssl_misc.h [hunk 1]
```diff
@@ -713,6 +713,7 @@ struct mbedtls_ssl_handshake_params {
     /*
      * Handshake specific crypto variables
      */
+
```
Comment (daverodgman) [inline library/ssl_misc.h:716]: Accidental whitespace change?

**Comment by tom-daubney-arm** [comment]
This PR is now being split into a series of smaller PRs in order to make reviewing easier. The first of these PRs is #7938 which contains the encrypt program.

**Comment by tom-daubney-arm** [comment]
I am now closing this PR since it has been broken down in to series of smaller PRs, as per this [comment](https://github.com/Mbed-TLS/mbedtls/pull/5064#issuecomment-1640016266).

**Pull request closed without merging**